geom/alg_densify_test.go

```diff
@@ -0,0 +1,75 @@
+package geom_test
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/peterstace/simplefeatures/geom"
+)
+
+func TestDensifyEmpty(t *testing.T) {
+	for _, empty := range []geom.Geometry{
+		geom.Point{}.AsGeometry(),
+		geom.LineString{}.AsGeometry(),
+		geom.Polygon{}.AsGeometry(),
+		geom.MultiPoint{}.AsGeometry(),
+		geom.MultiLineString{}.AsGeometry(),
+		geom.MultiPolygon{}.AsGeometry(),
+		geom.GeometryCollection{}.AsGeometry(),
+	} {
+		t.Run(empty.String(), func(t *testing.T) {
+			for _, ct := range []geom.CoordinatesType{
+				geom.DimXY,
+				geom.DimXYZ,
+				geom.DimXYM,
+				geom.DimXYZM,
+			} {
+				t.Run(ct.String(), func(t *testing.T) {
+					input := empty.ForceCoordinatesType(ct)
+					got := input.Densify(1.0)
+					expectGeomEq(t, got, input)
+				})
+			}
+		})
+	}
+}
+
+func TestDensify(t *testing.T) {
+	for i, tc := range []struct {
+		input   string
+		minDist float64
+		want    string
+	}{
+		// LineString with a single segment (tests threshold logic):
+		{"LINESTRING(0 0,1 0)", 2.0, "LINESTRING(0 0,1 0)"},
+		{"LINESTRING(0 0,1 0)", 1.0, "LINESTRING(0 0,1 0)"},
+		{"LINESTRING(0 0,1 0)", 0.9, "LINESTRING(0 0,0.5 0,1 0)"},
+		{"LINESTRING(0 0,1 0)", 0.5, "LINESTRING(0 0,0.5 0,1 0)"},
+		{"LINESTRING(0 0,1 0)", 0.4, "LINESTRING(0 0,0.3333333333333333 0,0.6666666666666666 0,1 0)"},
+		{"LINESTRING(0 0,1 0)", 0.3, "LINESTRING(0 0,0.25 0,0.5 0,0.75 0,1 0)"},
+		{"LINESTRING(0 0,1 0)", 0.25, "LINESTRING(0 0,0.25 0,0.5 0,0.75 0,1 0)"},
```

Comment:
Is having a `maxDist = 0` a realistic scenario that we should test/guard for?

Reply:
Ah! A user supplying weird values of `maxDist` is something I hadn't considered.

If a user supplies zero, then they will get an infinite loop. If a user provides a _very small_ max dist, the resultant geometry will just be very dense (it's easy to create a geometry with millions of control points). If a user supplies a negative max dist, then no additional control points are added (which feels weird).

We don't want to disallow very small but non-zero values of `maxDist artificially` (it would be too hard to get a reasonable cutoff in all use cases). However, we want to disallow zero and negative values.

557a096cf6b2ea30bd306128146fd81ac2f03cad detects the negative/zero cases and panics. That's reasonable behaviour since it would be a programmer error to supply a negative/zero value.